keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,252 @@
+
+### Prerequisites
+
+- [Java Development Kit (JDK)][jdk] with version 8 or above
+- [Azure Subscription][azure_subscription]
+- [Azure KeyVault][azure_keyvault]
+
+### Adding the package to your product
+
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-keyvault-secrets</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+</dependency>
+```
+
+### Create a Secret in the Azure Key Vault.
+
+To create an Azure Key Vault Store you can use the Azure Portal or [Azure CLI][azure_cli].
+
+Create the Azure Key Vault:
+```Powershell
+az keyvault create --name <keyvault-name> --resource-group <resource-group-name> --location eastus
+```
+
+After that, create the secret in the Azure Key Vault:
+```Powershell
+az keyvault secret set --vault-name <keyvault-name> --name "<secret-name>" --value "<secret-value>"
+```
+
+### Authenticate the client
+
+Applications that use a key vault must authenticate by using a token from Azure Active Directory. The owner of the application must register it in Azure Active Directory first. At the end of registration, the application owner gets the following values:
+
+* An Application ID (also known as the AAD Client ID or appID)
+* An authentication key (also known as the shared secret).
+
+The application must present both these values to Azure Active Directory, to get a token. 
+
+#### Create/Get Credentials
+
+Create an application in the Azure Active Directory.
+```Powershell
+az ad sp create-for-rbac -n <application-name> --password <application-password> --skip-assignment
+# If you don't specify a password, one will be created for you.
+```
+
+To authorize the same application to perform secret operarions in your vault, type the following command:
+```Powershell
+az keyvault set-policy --name <keyvault-name> --spn 8f8c4bbd-485b-45fd-98f7-ec6300b7b4ed --secret-permissions <secret-permissions>
+```
+
+#### Create Client
+
+In order to interact with the Azure Key Vault Secrets service you'll need to create an instance of the Secret Client class. To make this possible you'll need the application id and application key of an application in Azure Active Directory authorized with access to key vault.
+
+Once you have the values of the appliication id and application key you can create the secret client:
+
+```Java
+SecretClient client = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+```
+
+or
+
+```Java
+SecretAsyncClient client = SecretAsyncClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+```
+
+## Key concepts
+
+### Secret
+
+A secret is the fundamental resource within an Azure KeyVault. In its simplest form it is a name and a value. However, there are additional properties such as the modifiable content type and tags fields that allow the value to be interpreted or associated in different ways.
```

Comment:
consider putting properties in single quotes. i,e, The `expires` property ...

Reply:
Should we have them with bullet points? so it is easier to read?

Reply:
added bulltet points